core/server/models/post.js

```diff
@@ -61,39 +61,39 @@ Post = ghostBookshelf.Model.extend({
 
         this.on('created', function (model) {
             var isPage = !!model.get('page');
-            if (isPage) {
-                sitemap.pageAdded(model);
+
+                if (model.published) {
+                    events.emit(oldRT + '.unpublished', model);
+                }
+
+                events.emit(oldRT + '.deleted', model);
+                events.emit(newRT + '.added', model);
+
+                if (model.published) {
+                    events.emit(newRT + '.published', model);
+                }
             } else {
-                sitemap.postAdded(model);
-            }
-        });
-        this.on('updated', function (model) {
-            var isPage = !!model.get('page'),
-                wasPage = !!model.updated('page');
-
-            if (isPage && wasPage) {
-                // Page value didn't change, remains a page
-                sitemap.pageEdited(model);
-            } else if (!isPage && !wasPage) {
-                // Remains a Post
-                sitemap.postEdited(model);
-            } else if (isPage && !wasPage) {
-                // Switched from Post to Page
-                sitemap.postDeleted(model);
-                sitemap.pageAdded(model);
-            } else if (!isPage && wasPage) {
-                // Switched from Page to Post
-                sitemap.pageDeleted(model);
-                sitemap.postAdded(model);
+                if (model.statusChanging) {
+                    if (model.published) {
+                        events.emit(model.resourceType() + '.published', model);
+                    } else {
+                        events.emit(model.resourceType() + '.unpublished', model);
+                    }
+                } else if (!model.isNew()) {
+                    if (model.published) {
+                        events.emit(model.resourceType() + '.published.edited', model);
+                    }
+
+                    events.emit(model.resourceType() + '.edited', model);
```

Comment:
This is unquestionably the kind of logic that probably is wrong, or will turn out to be. I focused on getting publish/unpublish events right because they are going to be most important (less things care about draft posts).

Testing should help ensure the logic is correct, and then at the very least, it's all in one place and reusable by multiple components.